File: BloodFloater/src/BloodFloater.Models/Donor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace BloodFloater.Models
{
    public class Donor: BaseEntity
    {
        public int ProfileId { get; set; }

        public int DonationId { get; set; }

        public Profile Profile { get; set; }

        public virtual Donation Donation { get; set; }

        public virtual List<Donation> PastDonations { get; set; }
    }
}

File: BloodFloater/src/BloodFloater.Web.UI/wwwroot/src/app/models/donor.ts
import {Profile,Donation} from '../../app/models';

export class Donor{
    public ProfileId : number;
    public DonationId : number;

    public Profile : Profile;
    public Donation : Donation;

    public PastDonations : Array<Donation>;

    constructor(){
        this.Profile = new Profile();
        this.Donation = new Donation();

        this.PastDonations = new Array<Donation>();

    }
}